diff --git a/apps/Makefile b/apps/Makefile
index 8404d3482..379dbd096 100644
--- a/apps/Makefile
+++ b/apps/Makefile
@@ -1,3 +1,5 @@
+.NOTPARALLEL:
+
 ###########################################################################
 #
 # Copyright 2016-2017 Samsung Electronics All Rights Reserved.
diff --git a/apps/builtin/Makefile b/apps/builtin/Makefile
index 42d64f2ce..914700f20 100644
--- a/apps/builtin/Makefile
+++ b/apps/builtin/Makefile
@@ -1,3 +1,5 @@
+.NOTPARALLEL:
+
 ###########################################################################
 #
 # Copyright 2016 Samsung Electronics All Rights Reserved.
diff --git a/apps/examples/Makefile b/apps/examples/Makefile
index 9fa745697..045c0646b 100644
--- a/apps/examples/Makefile
+++ b/apps/examples/Makefile
@@ -1,3 +1,5 @@
+.NOTPARALLEL:
+
 ###########################################################################
 #
 # Copyright 2016-2017 Samsung Electronics All Rights Reserved.
diff --git a/apps/examples/board_specific_demo/Makefile b/apps/examples/board_specific_demo/Makefile
index 2c15acbcf..7ec027ac3 100644
--- a/apps/examples/board_specific_demo/Makefile
+++ b/apps/examples/board_specific_demo/Makefile
@@ -1,3 +1,5 @@
+.NOTPARALLEL:
+
 ###########################################################################
 #
 # Copyright 2021 Samsung Electronics All Rights Reserved.
diff --git a/apps/examples/board_specific_demo/ameba_cmsis_dsp_demo/Makefile b/apps/examples/board_specific_demo/ameba_cmsis_dsp_demo/Makefile
index 5a05d0171..5447f1b81 100644
--- a/apps/examples/board_specific_demo/ameba_cmsis_dsp_demo/Makefile
+++ b/apps/examples/board_specific_demo/ameba_cmsis_dsp_demo/Makefile
@@ -1,3 +1,5 @@
+.NOTPARALLEL:
+
 ###########################################################################
 #
 # Copyright 2024 Samsung Electronics All Rights Reserved.
diff --git a/apps/examples/board_specific_demo/ameba_cmsis_nn_demo/Makefile b/apps/examples/board_specific_demo/ameba_cmsis_nn_demo/Makefile
index 991d2e9c8..6d50ea43f 100644
--- a/apps/examples/board_specific_demo/ameba_cmsis_nn_demo/Makefile
+++ b/apps/examples/board_specific_demo/ameba_cmsis_nn_demo/Makefile
@@ -1,3 +1,5 @@
+.NOTPARALLEL:
+
 ###########################################################################
 #
 # Copyright 2023 Samsung Electronics All Rights Reserved.
diff --git a/apps/examples/dtls/Makefile b/apps/examples/dtls/Makefile
index 6159523b8..d048197de 100644
--- a/apps/examples/dtls/Makefile
+++ b/apps/examples/dtls/Makefile
@@ -1,3 +1,5 @@
+.NOTPARALLEL:
+
 ###########################################################################
 #
 # Copyright 2021 Samsung Electronics All Rights Reserved.
diff --git a/apps/examples/grpc/Makefile b/apps/examples/grpc/Makefile
index a0d2a7cdb..c3fdfae11 100644
--- a/apps/examples/grpc/Makefile
+++ b/apps/examples/grpc/Makefile
@@ -1,3 +1,5 @@
+.NOTPARALLEL:
+
 ###########################################################################
 #
 # Copyright 2021 Samsung Electronics All Rights Reserved.
diff --git a/apps/examples/iotivity_simpleserver/Makefile b/apps/examples/iotivity_simpleserver/Makefile
index 81bcbef0f..f09cea2e7 100644
--- a/apps/examples/iotivity_simpleserver/Makefile
+++ b/apps/examples/iotivity_simpleserver/Makefile
@@ -1,3 +1,5 @@
+.NOTPARALLEL:
+
 ############################################################################
 #
 # Copyright 2017 Samsung Electronics All Rights Reserved.
diff --git a/apps/examples/libcoap/Makefile b/apps/examples/libcoap/Makefile
index d16f406b3..74fc83178 100644
--- a/apps/examples/libcoap/Makefile
+++ b/apps/examples/libcoap/Makefile
@@ -1,3 +1,5 @@
+.NOTPARALLEL:
+
 ###########################################################################
 #
 # Copyright 2021 Samsung Electronics All Rights Reserved.
diff --git a/apps/examples/nettest/Makefile b/apps/examples/nettest/Makefile
index ed49d424e..d545b731d 100644
--- a/apps/examples/nettest/Makefile
+++ b/apps/examples/nettest/Makefile
@@ -1,3 +1,5 @@
+.NOTPARALLEL:
+
 ###########################################################################
 #
 # Copyright 2016 Samsung Electronics All Rights Reserved.
diff --git a/apps/examples/performance/Makefile b/apps/examples/performance/Makefile
index 9e1160d26..8e5f32564 100644
--- a/apps/examples/performance/Makefile
+++ b/apps/examples/performance/Makefile
@@ -1,3 +1,5 @@
+.NOTPARALLEL:
+
 ###########################################################################
 #
 # Copyright 2020 Samsung Electronics All Rights Reserved.
diff --git a/apps/examples/protobuf/Makefile b/apps/examples/protobuf/Makefile
index dc6c39767..9fc55d6eb 100644
--- a/apps/examples/protobuf/Makefile
+++ b/apps/examples/protobuf/Makefile
@@ -1,3 +1,5 @@
+.NOTPARALLEL:
+
 ###########################################################################
 #
 # Copyright 2021 Samsung Electronics All Rights Reserved.
diff --git a/apps/examples/security_test/Makefile b/apps/examples/security_test/Makefile
index bf4382746..7aef1e34c 100644
--- a/apps/examples/security_test/Makefile
+++ b/apps/examples/security_test/Makefile
@@ -1,3 +1,5 @@
+.NOTPARALLEL:
+
 ###########################################################################
 #
 # Copyright 2019 Samsung Electronics All Rights Reserved.
diff --git a/apps/examples/sensorbd_demo/Makefile b/apps/examples/sensorbd_demo/Makefile
index 89cff9c8b..35da35df5 100644
--- a/apps/examples/sensorbd_demo/Makefile
+++ b/apps/examples/sensorbd_demo/Makefile
@@ -1,3 +1,5 @@
+.NOTPARALLEL:
+
 ###########################################################################
 #
 # Copyright 2016 Samsung Electronics All Rights Reserved.
diff --git a/apps/examples/smartfs/Makefile b/apps/examples/smartfs/Makefile
index 9b9bd3029..5422976b0 100644
--- a/apps/examples/smartfs/Makefile
+++ b/apps/examples/smartfs/Makefile
@@ -1,3 +1,5 @@
+.NOTPARALLEL:
+
 ###########################################################################
 #
 # Copyright 2021 Samsung Electronics All Rights Reserved.
diff --git a/apps/examples/st_things/Makefile b/apps/examples/st_things/Makefile
index 449ed0740..06bbb1fc2 100644
--- a/apps/examples/st_things/Makefile
+++ b/apps/examples/st_things/Makefile
@@ -1,3 +1,5 @@
+.NOTPARALLEL:
+
 ###########################################################################
 #
 # Copyright 2017 Samsung Electronics All Rights Reserved.
diff --git a/apps/examples/testcase/Makefile b/apps/examples/testcase/Makefile
index 71dab4874..70b1444b0 100644
--- a/apps/examples/testcase/Makefile
+++ b/apps/examples/testcase/Makefile
@@ -1,3 +1,5 @@
+.NOTPARALLEL:
+
 ###########################################################################
 #
 # Copyright 2016 Samsung Electronics All Rights Reserved.
diff --git a/apps/examples/tls/Makefile b/apps/examples/tls/Makefile
index e8ab90a77..552a99994 100644
--- a/apps/examples/tls/Makefile
+++ b/apps/examples/tls/Makefile
@@ -1,3 +1,5 @@
+.NOTPARALLEL:
+
 ###########################################################################
 #
 # Copyright 2021 Samsung Electronics All Rights Reserved.
diff --git a/apps/examples/wifi_manager/Makefile b/apps/examples/wifi_manager/Makefile
index 591784143..af854154d 100644
--- a/apps/examples/wifi_manager/Makefile
+++ b/apps/examples/wifi_manager/Makefile
@@ -1,3 +1,5 @@
+.NOTPARALLEL:
+
 ###########################################################################
 #
 # Copyright 2021 Samsung Electronics All Rights Reserved.
diff --git a/apps/import/Makefile b/apps/import/Makefile
index bca501acb..0851e1c6f 100644
--- a/apps/import/Makefile
+++ b/apps/import/Makefile
@@ -1,3 +1,5 @@
+.NOTPARALLEL:
+
 ###########################################################################
 #
 # Copyright 2016 Samsung Electronics All Rights Reserved.
diff --git a/apps/platform/Makefile b/apps/platform/Makefile
index 89c41b17a..0f11739c2 100644
--- a/apps/platform/Makefile
+++ b/apps/platform/Makefile
@@ -1,3 +1,5 @@
+.NOTPARALLEL:
+
 ###########################################################################
 #
 # Copyright 2016-2017 Samsung Electronics All Rights Reserved.
diff --git a/apps/system/Makefile b/apps/system/Makefile
index d6057f04f..cf7f8bbcb 100644
--- a/apps/system/Makefile
+++ b/apps/system/Makefile
@@ -1,3 +1,5 @@
+.NOTPARALLEL:
+
 ###########################################################################
 #
 # Copyright 2016-2017 Samsung Electronics All Rights Reserved.
diff --git a/build/tools/esp32/Makefile b/build/tools/esp32/Makefile
index 3754a66bb..3d06d43cb 100644
--- a/build/tools/esp32/Makefile
+++ b/build/tools/esp32/Makefile
@@ -1,3 +1,5 @@
+.NOTPARALLEL:
+
 ###########################################################################
 # 
 # Copyright 2019 Samsung Electronics All Rights Reserved.
diff --git a/build/tools/esp32/esptool_py/Makefile b/build/tools/esp32/esptool_py/Makefile
index 23cb9c66d..66d9fd12a 100644
--- a/build/tools/esp32/esptool_py/Makefile
+++ b/build/tools/esp32/esptool_py/Makefile
@@ -1,3 +1,5 @@
+.NOTPARALLEL:
+
 ###########################################################################
 #
 # Copyright 2019 Samsung Electronics All Rights Reserved.
diff --git a/build/tools/openocd/source/Makefile b/build/tools/openocd/source/Makefile
index 0f6b76435..ef7ea7fe3 100644
--- a/build/tools/openocd/source/Makefile
+++ b/build/tools/openocd/source/Makefile
@@ -1,3 +1,5 @@
+.NOTPARALLEL:
+
 ###########################################################################
 #
 # Copyright 2017 Samsung Electronics All Rights Reserved.
diff --git a/external/Makefile b/external/Makefile
index 14651e96a..1b314cbc1 100644
--- a/external/Makefile
+++ b/external/Makefile
@@ -1,3 +1,5 @@
+.NOTPARALLEL:
+
 ############################################################################
 #
 # Copyright 2016 Samsung Electronics All Rights Reserved.
@@ -104,7 +106,7 @@ $(CXXOBJS): %$(OBJEXT): %.cpp
 
 define SDIR_template
 $(1)_$(2):
-	$(Q) $(MAKE) -C $(1) $(2) TOPDIR="$(TOPDIR)" EXTDIR="$(EXTDIR)"
+	$(Q) +$(MAKE) -C $(1) $(2) TOPDIR="$(TOPDIR)" EXTDIR="$(EXTDIR)"
 
 endef
 
diff --git a/external/WICED/Makefile b/external/WICED/Makefile
index 4ce74786a..91b308d36 100644
--- a/external/WICED/Makefile
+++ b/external/WICED/Makefile
@@ -1,3 +1,5 @@
+.NOTPARALLEL:
+
 ############################################################################
 #
 # Copyright 2018, Cypress Semiconductor Corporation or a subsidiary of 
diff --git a/external/avs/Makefile b/external/avs/Makefile
index bc82e1d0f..80d15ed9c 100644
--- a/external/avs/Makefile
+++ b/external/avs/Makefile
@@ -1,3 +1,5 @@
+.NOTPARALLEL:
+
 ###########################################################################
 #
 # Copyright 2016 Samsung Electronics All Rights Reserved.
diff --git a/external/aws/Makefile b/external/aws/Makefile
index 0ab315476..ad36339ca 100644
--- a/external/aws/Makefile
+++ b/external/aws/Makefile
@@ -1,3 +1,5 @@
+.NOTPARALLEL:
+
 ###########################################################################
 #
 # Copyright 2017 Samsung Electronics All Rights Reserved.
diff --git a/external/cmsis_dsp/Makefile b/external/cmsis_dsp/Makefile
index 12fb3d2ef..9c83e650c 100644
--- a/external/cmsis_dsp/Makefile
+++ b/external/cmsis_dsp/Makefile
@@ -1,3 +1,5 @@
+.NOTPARALLEL:
+
 ###########################################################################
 #
 # Copyright 2024 Samsung Electronics All Rights Reserved.
diff --git a/external/error_report/Makefile b/external/error_report/Makefile
index c1e31b3d3..aa19856ad 100644
--- a/external/error_report/Makefile
+++ b/external/error_report/Makefile
@@ -1,3 +1,5 @@
+.NOTPARALLEL:
+
 ###########################################################################
 #
 # Copyright 2018 Samsung Electronics All Rights Reserved.
diff --git a/external/esp_idf_port/Makefile b/external/esp_idf_port/Makefile
index d6f49368d..53dac7791 100644
--- a/external/esp_idf_port/Makefile
+++ b/external/esp_idf_port/Makefile
@@ -1,3 +1,5 @@
+.NOTPARALLEL:
+
 ###########################################################################
 #
 # Copyright 2019 Samsung Electronics All Rights Reserved.
diff --git a/external/esp_idf_port/esp32/Makefile b/external/esp_idf_port/esp32/Makefile
index a3963a620..908ae9a26 100644
--- a/external/esp_idf_port/esp32/Makefile
+++ b/external/esp_idf_port/esp32/Makefile
@@ -1,3 +1,5 @@
+.NOTPARALLEL:
+
 -include ${TOPDIR}/.config
 include ${TOPDIR}/tools/Config.mk
 include ${TOPDIR}/arch/xtensa/src/lx6/Toolchain.defs
diff --git a/external/gmock/Makefile b/external/gmock/Makefile
index 91b6fdae4..609820e5a 100644
--- a/external/gmock/Makefile
+++ b/external/gmock/Makefile
@@ -1,3 +1,5 @@
+.NOTPARALLEL:
+
 ###########################################################################
 #
 # Copyright 2018 Samsung Electronics All Rights Reserved.
diff --git a/external/grpc/Makefile b/external/grpc/Makefile
index 42a5b86d9..7239fecb5 100644
--- a/external/grpc/Makefile
+++ b/external/grpc/Makefile
@@ -1,3 +1,5 @@
+.NOTPARALLEL:
+
 ###########################################################################
 #
 # Copyright 2016 Samsung Electronics All Rights Reserved.
diff --git a/external/iotivity/iotivity_1.2-rel/extlibs/gtest/gtest-1.7.0/Makefile b/external/iotivity/iotivity_1.2-rel/extlibs/gtest/gtest-1.7.0/Makefile
index cf35458be..5d58b4d4e 100644
--- a/external/iotivity/iotivity_1.2-rel/extlibs/gtest/gtest-1.7.0/Makefile
+++ b/external/iotivity/iotivity_1.2-rel/extlibs/gtest/gtest-1.7.0/Makefile
@@ -1,3 +1,5 @@
+.NOTPARALLEL:
+
 # Makefile.in generated by automake 1.11.3 from Makefile.am.
 # Makefile.  Generated from Makefile.in by configure.
 
diff --git a/external/iotivity/iotivity_1.2-rel/extlibs/tinycbor/tinycbor/Makefile b/external/iotivity/iotivity_1.2-rel/extlibs/tinycbor/tinycbor/Makefile
index e21b5507d..98f1afedf 100644
--- a/external/iotivity/iotivity_1.2-rel/extlibs/tinycbor/tinycbor/Makefile
+++ b/external/iotivity/iotivity_1.2-rel/extlibs/tinycbor/tinycbor/Makefile
@@ -1,3 +1,5 @@
+.NOTPARALLEL:
+
 # Variables:
 prefix = /usr/local
 exec_prefix = $(prefix)
diff --git a/external/iotivity/iotivity_1.2-rel/resource/csdk/connectivity/build/android/Makefile b/external/iotivity/iotivity_1.2-rel/resource/csdk/connectivity/build/android/Makefile
index 038e36bdc..43a013257 100644
--- a/external/iotivity/iotivity_1.2-rel/resource/csdk/connectivity/build/android/Makefile
+++ b/external/iotivity/iotivity_1.2-rel/resource/csdk/connectivity/build/android/Makefile
@@ -1,3 +1,5 @@
+.NOTPARALLEL:
+
 #Assumes ndk directory is at ~/
 #set ndk directory PATH in .bashrc and use the ndk-build directly
 #change NDK_PATH to ${your_ndk_path}
diff --git a/external/iotivity/iotivity_1.2-rel/resource/csdk/connectivity/build/tizen/Makefile b/external/iotivity/iotivity_1.2-rel/resource/csdk/connectivity/build/tizen/Makefile
index 287ec5ff8..092e738fd 100644
--- a/external/iotivity/iotivity_1.2-rel/resource/csdk/connectivity/build/tizen/Makefile
+++ b/external/iotivity/iotivity_1.2-rel/resource/csdk/connectivity/build/tizen/Makefile
@@ -1,3 +1,5 @@
+.NOTPARALLEL:
+
 ##
 # override with `make BUILD=debug`
 # override with `make PLATFORM=arduinomega` or `make PLATFORM=arduinodue`
diff --git a/external/iotjs/config/tizenrt/Makefile b/external/iotjs/config/tizenrt/Makefile
index 42b199b04..d3620da2b 100644
--- a/external/iotjs/config/tizenrt/Makefile
+++ b/external/iotjs/config/tizenrt/Makefile
@@ -1,3 +1,5 @@
+.NOTPARALLEL:
+
 ###########################################################################
 #
 # Copyright 2018 Samsung Electronics All Rights Reserved.
diff --git a/external/iotjs/deps/http-parser/Makefile b/external/iotjs/deps/http-parser/Makefile
index 62009ffdd..484823d2c 100644
--- a/external/iotjs/deps/http-parser/Makefile
+++ b/external/iotjs/deps/http-parser/Makefile
@@ -1,3 +1,5 @@
+.NOTPARALLEL:
+
 # Copyright Joyent, Inc. and other Node contributors. All rights reserved.
 #
 # Permission is hereby granted, free of charge, to any person obtaining a copy
diff --git a/external/iotjs/deps/jerry/targets/esp8266/Makefile b/external/iotjs/deps/jerry/targets/esp8266/Makefile
index f034f1f63..d540176a3 100644
--- a/external/iotjs/deps/jerry/targets/esp8266/Makefile
+++ b/external/iotjs/deps/jerry/targets/esp8266/Makefile
@@ -1,3 +1,5 @@
+.NOTPARALLEL:
+
 # Copyright JS Foundation and other contributors, http://js.foundation
 #
 # Licensed under the Apache License, Version 2.0 (the "License");
diff --git a/external/iotjs/deps/jerry/targets/mbedos5/Makefile b/external/iotjs/deps/jerry/targets/mbedos5/Makefile
index faf130c96..b7d67b4e1 100644
--- a/external/iotjs/deps/jerry/targets/mbedos5/Makefile
+++ b/external/iotjs/deps/jerry/targets/mbedos5/Makefile
@@ -1,3 +1,5 @@
+.NOTPARALLEL:
+
 # Copyright JS Foundation and other contributors, http://js.foundation
 #
 # Licensed under the Apache License, Version 2.0 (the "License");
diff --git a/external/iotjs/deps/jerry/targets/riot-stm32f4/Makefile b/external/iotjs/deps/jerry/targets/riot-stm32f4/Makefile
index a1ce8083f..1a2777f12 100644
--- a/external/iotjs/deps/jerry/targets/riot-stm32f4/Makefile
+++ b/external/iotjs/deps/jerry/targets/riot-stm32f4/Makefile
@@ -1,3 +1,5 @@
+.NOTPARALLEL:
+
 # Copyright JS Foundation and other contributors, http://js.foundation
 #
 # Licensed under the Apache License, Version 2.0 (the "License");
diff --git a/external/iotjs/deps/jerry/targets/zephyr/Makefile b/external/iotjs/deps/jerry/targets/zephyr/Makefile
index f053626a3..1d0359a79 100644
--- a/external/iotjs/deps/jerry/targets/zephyr/Makefile
+++ b/external/iotjs/deps/jerry/targets/zephyr/Makefile
@@ -1,3 +1,5 @@
+.NOTPARALLEL:
+
 # Copyright JS Foundation and other contributors, http://js.foundation
 #
 # Licensed under the Apache License, Version 2.0 (the "License");
diff --git a/external/iotjs/deps/libtuv/Makefile b/external/iotjs/deps/libtuv/Makefile
index 1ab6509e2..af60135e9 100644
--- a/external/iotjs/deps/libtuv/Makefile
+++ b/external/iotjs/deps/libtuv/Makefile
@@ -1,3 +1,5 @@
+.NOTPARALLEL:
+
 # Copyright 2015-2017 Samsung Electronics Co., Ltd.
 #
 # Licensed under the Apache License, Version 2.0 (the "License");
diff --git a/external/libcxx-test/Makefile b/external/libcxx-test/Makefile
index 8f64bf134..892fb89e2 100644
--- a/external/libcxx-test/Makefile
+++ b/external/libcxx-test/Makefile
@@ -1,3 +1,5 @@
+.NOTPARALLEL:
+
 ###########################################################################
 #
 # Copyright 2018 Samsung Electronics All Rights Reserved.
diff --git a/external/libopus/Makefile b/external/libopus/Makefile
index 0bf2a4b42..fa7ab3248 100644
--- a/external/libopus/Makefile
+++ b/external/libopus/Makefile
@@ -1,3 +1,5 @@
+.NOTPARALLEL:
+
 ###########################################################################
 #
 # Copyright 2018 Samsung Electronics All Rights Reserved.
diff --git a/external/libsodium/Makefile b/external/libsodium/Makefile
index 7c07ababd..de62b59a6 100644
--- a/external/libsodium/Makefile
+++ b/external/libsodium/Makefile
@@ -1,3 +1,5 @@
+.NOTPARALLEL:
+
 ###########################################################################
 #
 # Copyright 2019 Samsung Electronics All Rights Reserved.
diff --git a/external/libtuv/Makefile b/external/libtuv/Makefile
index 772caa1a7..9e4ccf21d 100644
--- a/external/libtuv/Makefile
+++ b/external/libtuv/Makefile
@@ -1,3 +1,5 @@
+.NOTPARALLEL:
+
 ###########################################################################
 #
 # Copyright 2017 Samsung Electronics All Rights Reserved.
diff --git a/external/nanopb/Makefile b/external/nanopb/Makefile
index 929d1bce7..2adb35933 100644
--- a/external/nanopb/Makefile
+++ b/external/nanopb/Makefile
@@ -1,3 +1,5 @@
+.NOTPARALLEL:
+
 ###########################################################################
 #
 # Copyright 2019 Samsung Electronics All Rights Reserved.
diff --git a/external/protobuf/Makefile b/external/protobuf/Makefile
index 350b4ff00..58057a38d 100644
--- a/external/protobuf/Makefile
+++ b/external/protobuf/Makefile
@@ -1,3 +1,5 @@
+.NOTPARALLEL:
+
 ###########################################################################
 #
 # Copyright 2016 Samsung Electronics All Rights Reserved.
diff --git a/external/stdk/Makefile b/external/stdk/Makefile
index a307cb146..f3a795201 100644
--- a/external/stdk/Makefile
+++ b/external/stdk/Makefile
@@ -1,3 +1,5 @@
+.NOTPARALLEL:
+
 ###########################################################################
 #
 # Copyright 2019 Samsung Electronics All Rights Reserved.
diff --git a/external/swepd/Makefile b/external/swepd/Makefile
index 7bc29aab2..8c7e607de 100644
--- a/external/swepd/Makefile
+++ b/external/swepd/Makefile
@@ -1,3 +1,5 @@
+.NOTPARALLEL:
+
 ###########################################################################
 #
 # Copyright 2018 Samsung Electronics All Rights Reserved.
diff --git a/external/tfmicro/Makefile b/external/tfmicro/Makefile
index 5a8e418e9..b17fb325a 100644
--- a/external/tfmicro/Makefile
+++ b/external/tfmicro/Makefile
@@ -1,3 +1,5 @@
+.NOTPARALLEL:
+
 ###########################################################################
 #
 # Copyright 2023 Samsung Electronics All Rights Reserved.
diff --git a/external/wakaama/Makefile b/external/wakaama/Makefile
index 1e26f26fe..f21ca050b 100644
--- a/external/wakaama/Makefile
+++ b/external/wakaama/Makefile
@@ -1,3 +1,5 @@
+.NOTPARALLEL:
+
 ###########################################################################
 #
 # Copyright 2017 Samsung Electronics All Rights Reserved.
diff --git a/external/websocket/Makefile b/external/websocket/Makefile
index 323d7880e..205a3662b 100644
--- a/external/websocket/Makefile
+++ b/external/websocket/Makefile
@@ -1,3 +1,5 @@
+.NOTPARALLEL:
+
 ###########################################################################
 #
 # Copyright 2016 Samsung Electronics All Rights Reserved.
diff --git a/external/wpa_supplicant/Makefile b/external/wpa_supplicant/Makefile
index c17ee924b..f47bdfbaa 100644
--- a/external/wpa_supplicant/Makefile
+++ b/external/wpa_supplicant/Makefile
@@ -1,3 +1,5 @@
+.NOTPARALLEL:
+
 ###########################################################################
 #
 # Copyright 2017 Samsung Electronics All Rights Reserved.
diff --git a/external/wpa_supplicant/src/Makefile b/external/wpa_supplicant/src/Makefile
index 10e0171b0..7284e4a50 100644
--- a/external/wpa_supplicant/src/Makefile
+++ b/external/wpa_supplicant/src/Makefile
@@ -1,3 +1,5 @@
+.NOTPARALLEL:
+
 SUBDIRS=ap common crypto drivers eapol_auth eapol_supp eap_common eap_peer eap_server l2_packet p2p pae radius rsn_supp tls utils wps
 
 all:
diff --git a/loadable_apps/Makefile b/loadable_apps/Makefile
index ee0e5ad1c..62bb2dddc 100644
--- a/loadable_apps/Makefile
+++ b/loadable_apps/Makefile
@@ -1,3 +1,5 @@
+.NOTPARALLEL:
+
 ###########################################################################
 #
 # Copyright 2019 Samsung Electronics All Rights Reserved.
diff --git a/loadable_apps/loadable_sample/Makefile b/loadable_apps/loadable_sample/Makefile
index 911c9e640..f76c1ad07 100644
--- a/loadable_apps/loadable_sample/Makefile
+++ b/loadable_apps/loadable_sample/Makefile
@@ -1,3 +1,5 @@
+.NOTPARALLEL:
+
 ###########################################################################
 #
 # Copyright 2019 Samsung Electronics All Rights Reserved.
diff --git a/os/Makefile b/os/Makefile
index c643fcc98..e54ea3f5b 100644
--- a/os/Makefile
+++ b/os/Makefile
@@ -1,3 +1,5 @@
+.NOTPARALLEL:
+
 ###########################################################################
 #
 # Copyright 2016 Samsung Electronics All Rights Reserved.
diff --git a/os/arch/xtensa/src/Makefile b/os/arch/xtensa/src/Makefile
index d7258969b..d080135c6 100644
--- a/os/arch/xtensa/src/Makefile
+++ b/os/arch/xtensa/src/Makefile
@@ -1,3 +1,5 @@
+.NOTPARALLEL:
+
 ############################################################################
 #
 #Copyright 2018 Samsung Electronics All Rights Reserved.
diff --git a/os/binfmt/Makefile b/os/binfmt/Makefile
index 1b58e089e..6b6fccc9d 100644
--- a/os/binfmt/Makefile
+++ b/os/binfmt/Makefile
@@ -1,3 +1,5 @@
+.NOTPARALLEL:
+
 ###########################################################################
 #
 # Copyright 2019 Samsung Electronics All Rights Reserved.
diff --git a/os/board/imxrt1020-evk/src/Makefile b/os/board/imxrt1020-evk/src/Makefile
index fcb10ae9c..cf55b5999 100644
--- a/os/board/imxrt1020-evk/src/Makefile
+++ b/os/board/imxrt1020-evk/src/Makefile
@@ -1,3 +1,5 @@
+.NOTPARALLEL:
+
 ###########################################################################
 #
 # Copyright 2019 NXP Semiconductors All Rights Reserved.
diff --git a/os/board/imxrt1050-evk/src/Makefile b/os/board/imxrt1050-evk/src/Makefile
index 6a49072fc..5f88cf514 100644
--- a/os/board/imxrt1050-evk/src/Makefile
+++ b/os/board/imxrt1050-evk/src/Makefile
@@ -1,3 +1,5 @@
+.NOTPARALLEL:
+
 ###########################################################################
 #
 # Copyright 2019 NXP Semiconductors All Rights Reserved.
diff --git a/os/board/rtl8720e/src/Makefile b/os/board/rtl8720e/src/Makefile
index 1da0432f1..9333d249b 100644
--- a/os/board/rtl8720e/src/Makefile
+++ b/os/board/rtl8720e/src/Makefile
@@ -1,3 +1,5 @@
+.NOTPARALLEL:
+
 ###########################################################################
 #
 # Copyright 2022 Samsung Electronics All Rights Reserved.
diff --git a/os/board/rtl8721csm/src/Makefile b/os/board/rtl8721csm/src/Makefile
index 8aaae455c..561c455bd 100644
--- a/os/board/rtl8721csm/src/Makefile
+++ b/os/board/rtl8721csm/src/Makefile
@@ -1,3 +1,5 @@
+.NOTPARALLEL:
+
 ###########################################################################
 #
 # Copyright 2020 Samsung Electronics All Rights Reserved.
diff --git a/os/compression/Makefile b/os/compression/Makefile
index ec5a7ef4b..6ec34e944 100644
--- a/os/compression/Makefile
+++ b/os/compression/Makefile
@@ -1,3 +1,5 @@
+.NOTPARALLEL:
+
 ###########################################################################
 #
 # Copyright 2019 Samsung Electronics All Rights Reserved.
diff --git a/os/dbuild.sh b/os/dbuild.sh
index 2a3916be3..84fc615d1 100755
--- a/os/dbuild.sh
+++ b/os/dbuild.sh
@@ -29,7 +29,7 @@ DOCKER_PUBLIC_IMAGE="tizenrt/tizenrt"
 DOCKER_VERSION="1.5.8"
 
 STATUS_LIST="NOT_CONFIGURED BOARD_CONFIGURED CONFIGURED BUILT PREPARE_DL DOWNLOAD_READY"
-BUILD_CMD=make
+BUILD_CMD="make -j$(nproc)"
 
 # Checking docker is installed
 nodocker() {
diff --git a/os/drivers/wireless/realtek/os/os_dep/Makefile b/os/drivers/wireless/realtek/os/os_dep/Makefile
index 1a646a9d8..c2cdb6071 100644
--- a/os/drivers/wireless/realtek/os/os_dep/Makefile
+++ b/os/drivers/wireless/realtek/os/os_dep/Makefile
@@ -1,3 +1,5 @@
+.NOTPARALLEL:
+
 
 include $(MAKE_INCLUDE_GEN)
 
diff --git a/os/se/Makefile b/os/se/Makefile
index 44669c658..c3c352e66 100644
--- a/os/se/Makefile
+++ b/os/se/Makefile
@@ -1,3 +1,5 @@
+.NOTPARALLEL:
+
 ############################################################################
 #
 # Copyright 2019 Samsung Electronics All Rights Reserved.
diff --git a/os/syscall/Makefile b/os/syscall/Makefile
index 346d0d1a6..64c530f1e 100644
--- a/os/syscall/Makefile
+++ b/os/syscall/Makefile
@@ -1,3 +1,5 @@
+.NOTPARALLEL:
+
 ###########################################################################
 #
 # Copyright 2016 Samsung Electronics All Rights Reserved.
diff --git a/os/tools/compression/Makefile b/os/tools/compression/Makefile
index 876488c8f..8207372fa 100644
--- a/os/tools/compression/Makefile
+++ b/os/tools/compression/Makefile
@@ -1,3 +1,5 @@
+.NOTPARALLEL:
+
 ###########################################################################
 #
 # Copyright 2019 Samsung Electronics All Rights Reserved.
diff --git a/tools/Makefile b/tools/Makefile
index 7b166b1f9..e6e61ba8d 100644
--- a/tools/Makefile
+++ b/tools/Makefile
@@ -1,3 +1,5 @@
+.NOTPARALLEL:
+
 ###########################################################################
 #
 # Copyright 2016 Samsung Electronics All Rights Reserved.
diff --git a/tools/araui/sim/template/Makefile b/tools/araui/sim/template/Makefile
index 041323ea6..049a929de 100644
--- a/tools/araui/sim/template/Makefile
+++ b/tools/araui/sim/template/Makefile
@@ -1,3 +1,5 @@
+.NOTPARALLEL:
+
 include araui.mk
 
 TARGET = sim
diff --git a/tools/nxfuse/Makefile b/tools/nxfuse/Makefile
index 0933ee672..54b366d8f 100644
--- a/tools/nxfuse/Makefile
+++ b/tools/nxfuse/Makefile
@@ -1,3 +1,5 @@
+.NOTPARALLEL:
+
 ###########################################################################
 #
 # Copyright 2019 Samsung Electronics All Rights Reserved.
